VideoLocker/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -51,20 +52,21 @@ public boolean isDownloadOverWifiOnly() {
      */
     public File getDownloadFolder() {
         ProfileModel profile = getProfile();
-        
-        File android = new File(Environment.getExternalStorageDirectory(), "Android");
-        File downloadsDir = new File(android, "data");
-        File packDir = new File(downloadsDir, context.getPackageName());
-        File edxDir = new File(packDir, profile.username);
-        edxDir.mkdirs();
+        File videosDir = new File(context.getExternalFilesDir(null).getParent(), AppConstants.Folders.VIDEOS);
+        File usersVidDir = new File(videosDir, Sha1Util.SHA1(profile.username));
+        if (!usersVidDir.exists()) {
+            usersVidDir.mkdirs();
+        }
         try {
-            File noMediaFile = new File(edxDir, ".nomedia");
-            noMediaFile.createNewFile();
-        }catch (IOException ioException){
+            File noMediaFile = new File(usersVidDir, ".nomedia");
+            if (!noMediaFile.exists()) {
+                noMediaFile.createNewFile();
+            }
```

Comment:
The `exists()` check is not needed here either. The `createNewFile()` method also simply returns `false` if the file already exists.